cmd/tools/doctor/doctor.go

```diff
@@ -0,0 +1,177 @@
+package doctor
+
+import (
+	"fmt"
+	"github.com/spf13/cobra"
+	"gopkg.in/yaml.v3"
+	"io/ioutil"
+	"os"
+)
+
+type options struct {
+	ShouldPrintConfig bool
+}
+
+var opts = &options{
+	ShouldPrintConfig: false,
+}
+
+var Cmd = &cobra.Command{
+	Use:   "doctor",
+	Short: "Check for potential problems",
+	Long:  "Check for potential problems",
+	Run:   doDoctorCmd,
+}
+
+func doDoctorCmd(cmd *cobra.Command, _ []string) {
+	var config = cmd.Root().PersistentFlags().Lookup("config")
+	doDoctor(config.Value.String())
+}
+
+// Pointers need to be used for struct members where you need
+// to distinguish missing values from zero values. See
+// https://github.com/go-yaml/yaml/issues/113 for details
+// The downside of making all members pointers is accessing
+// the values requires more dereferencing - see doctor_test.go
+
+type Consul struct {
+	Host        *string   `yaml:"host,omitempty"`
+	ServiceName *string   `yaml:"service_name,omitempty"`
+	Tags        *[]string `yaml:"tags,omitempty"`
+}
+
+type Tools struct {
```

Comment:
i think all these structs and yml parsing shall be defined in other go file.  May be only keep harvestdoctor related methods here?